src/WebJobs.Script.WebHost/Management/LinuxSpecialization/IAssignmentHandler.cs

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.WebHost.Models;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Management.LinuxSpecialization
+{
+    public interface IAssignmentHandler
```

Comment:
While I like the direction you are going with reducing duplicate code, I am still conflicted regarding this abstraction layer. I _think_ what would be a better fit would be to derive a `InstanceManagerBase` class which has the common logic shared between the two, then appropriate `protected abstract` methods for the hooks where logic needs to be customized. The legion version can have almost all of those hooks be a no-op, and the atlas version can hold its current logic.

Reply:
Realize now @fabiocav suggested essentially the same thing.

Reply:
@fabiocav @jviau I think your suggestion is a better approach since the InstanceManager class doesn't have much logic after introducing a new IAssignmentHandler interface. I have [another PR that utilizes a different InstanceManager](https://github.com/Azure/azure-functions-host/pull/8634) (one for Atlas and one for Legion) like you guys have suggested.
I was looking through the InstanceManager class, and I found that _GetInstanceInfo()_ and _Reset()_ (Internal method) are the only methods with the exact same implementation. So, I don't think an abstract class is necessary for this case. Should I just keep the changes in the [other PR](https://github.com/Azure/azure-functions-host/pull/8634) as is?

Thank you for your suggestions!